src/wp-includes/widgets.php

```diff
@@ -1327,7 +1337,11 @@ function sync_registered_widgets( $theme_changed = false ) {
 	$sidebars_widgets['wp_inactive_widgets'] = array_merge( $lost_widgets, (array) $sidebars_widgets['wp_inactive_widgets'] );
 
 	if ( 'customize' !== $theme_changed ) {
-		wp_set_sidebars_widgets( $sidebars_widgets );
+		/*
+		 * The second parameter is false to avoid recursion: refreshing the global
+		 * $sidebars_widgets entails calling retrieve_widgets().
```

Comment:
This references the old function name, `retrieve_widgets()`.

Reply:
It seems like the name change may not go through so I will leave it that way.